### PR TITLE
Check if typeof Proxy is a function

### PR DESCRIPTION
**Pull request by christianbach**
Since Proxy is only supported above iOS 10 it will throw an error on iOS 8 and iOS 9. And if one would implement a custom JSC for android the current detection will not work. Changes this to check Proxy with typeof solved both scenarios. Thanks for a great library!

### CODE REVIEW

File: src/core/AnimatedEvent.js [hunk 1]
```diff
@@ -58,9 +58,9 @@ function sanitizeArgMapping(argMapping) {
     };
 
     const proxy =
-      Platform.OS === 'android'
-        ? createEventObjectProxyPolyfill()
-        : new Proxy({}, proxyHandler);
+      typeof Proxy === 'function'
```
Comment (osdnk) [inline src/core/AnimatedEvent.js]:
```suggestion
      Proxy
```

@christianbach / cc @kmagiera 
How about this change? `Proxy` on Android is `undefined` I suppose

Comment (christianbach) [reply src/core/AnimatedEvent.js]: Thats correct Proxy is undefined on "Android" with the JSC shipping with react-native and on iOS below 10

Comment (osdnk) [reply src/core/AnimatedEvent.js]: nice, so I believe that my suggestion could be applied. It makes it more readable I suppose

Comment (christianbach) [reply src/core/AnimatedEvent.js]: 👍 I will implement both your suggestion and @kmagiera and update the PR. Thanks!

File: src/ConfigHelper.js [hunk 2]
```diff
@@ -73,16 +73,11 @@ let NATIVE_THREAD_PROPS_WHITELIST = {
   borderBottomLeftRadius: true,
   borderBottomRightRadius: true,
   borderBottomStartRadius: true,
-  borderBottomWidth: true,
-  borderLeftWidth: true,
   borderRadius: true,
-  borderRightWidth: true,
```
Comment (osdnk) [inline src/ConfigHelper.js]: thanks

File: src/core/createEventObjectProxyPolyfill.js [hunk 3]
```diff
@@ -33,7 +34,7 @@ export default function createEventObjectProxyPolyfill() {
     contentInset: { right: {}, top: {}, left: {}, bottom: {} },
   };
   const traverse = obj => {
-    for (key in obj) {
```
Comment (osdnk) [inline src/core/createEventObjectProxyPolyfill.js]: 😅

File: src/derived/evaluateOnce.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import AnimatedValue from '../core/AnimatedValue';
-import { block, call, always, cond, set } from '../base';
```
Comment (osdnk) [inline src/derived/evaluateOnce.js]: 🤷‍♂️

File: src/derived/onChange.js [hunk 5]
```diff
@@ -5,6 +5,6 @@ export default function onChange(value, action) {
   const prevValue = new AnimatedValue();
   return block([
     cond(not(defined(prevValue)), set(prevValue, value)),
-    cond(neq(value, prevValue), [set(prevValue, value), action])
```
Comment (osdnk) [inline src/derived/onChange.js]: 👍

File: src/core/AnimatedEvent.js (hunk 1 above)
Comment (christianbach) [reply src/core/AnimatedEvent.js]: Done, updated the PR with the suggested changes and added a bonus (linted and formated the code base)

**Pull request merged**